H/HDF4/build_tarballs.jl

```diff
@@ -0,0 +1,57 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "HDF4"
+version = v"2.16.0"
```

Comment:
Isn't the version number 
```suggestion
version = v"4.2.16"
```
?

Reply:
I am not sure about this, since we are explicitly talking about HDF4 which I would consider to be a completely different data format compared to HDF5, and not just a different version. Also, practically I don't know if we can assume backwards compatibility if ever a new minor version gets released (which I think will not happen).

Reply:
I opened the tarball and saw references to the version number 4.2.16 everywhere.  Also Linux distributions call it 4.2.X, for example Debian https://packages.debian.org/source/sid/libhdf4, and all other distributions: https://repology.org/project/hdf/versions